### PR TITLE
feat(flow): NLU in what-next questions (#342)

### LINKED ISSUE #342: Add NLU in "what next" questions that fallback on QA
Most of these should require a simple addition to the unsupported intent policy, and additions/modifications to stories
To manage get_assessment intent, we could add a generic condition to the policy to check on the self-assess-done slot value and goes in the fallback if it is true
This might allow to reduce the number of stories associated with variations on message that are only one option (the need_assessment cases will remain)

*If some cases require more adjustments than these, they should probably be left out to another issue

### PR DESCRIPTION
**Pull request by NetBKarine**
## Description

Follows the new flows with NLU, some places still have buttons to guide the user, but free text is enabled

## Related issues

closes #342 

## Checklist

- [X] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [X] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: core/domain/domain.en.yml [hunk 1]
```diff
@@ -1310,33 +1238,30 @@ responses:
...
+  utter_ask_test_navigation__continue:
     - text: Do you want to continue and search for a testing site?
+
+  utter_ask_test_navigation__continue_error_offer_assessment:
+    - text: Sorry, do you want to continue and search for a testing site?
       buttons:
-        - title: Yes continue
-          payload: /continue
+        - title: Yes
+          payload: "/affirm"
```
Comment (fparga) [inline core/domain/domain.en.yml:1248]: Is there a reason to have some payloads with quotes and others without?

File: core/core/policies/unsupported_intent_policy.py [hunk 2]
```diff
@@ -20,11 +34,33 @@
         "get_assessment",
         "tested_positive",
         "navigate_test_locations",
-        "fallback",
+    ],
+    "utter_ask_test_navigation__continue": ["done", "get_assessment", "ask_question"]
+    + AFFIRM_DENY_INTENTS,
```
[the remark this answers is not in the document]
Comment (NetBKarine) [reply core/core/policies/unsupported_intent_policy.py:39]: I didn't write it this way it was corrected by one of the formatters...

File: action-server/tests/actions/test_test_navigation_form.py [hunk 3]
```diff
@@ -567,11 +542,7 @@ def test_postal_code_many_results(self, mock_maps):
                 SlotSet(LOCATIONS_SLOT, [TESTING_LOCATION_RAW, TESTING_LOCATION_RAW]),
             ]
             + CLEARED_SLOTS
-            + [
-                SlotSet(TEST_NAVIGATION_SUCCESS_SLOT, True),
```
[the remark this answers is not in the document]
Comment (NetBKarine) [reply action-server/tests/actions/test_test_navigation_form.py:571]: It was a featurized slot used in stories to ask the correct what next question. The design changed and the question is now the same whether there was an error or not, so it became useless

File: core/domain/domain.en.yml (hunk 1 above)
Comment (NetBKarine) [reply core/domain/domain.en.yml:1248]: There's a reason but it's not a good one. Both are valid we just didn't chose a format (no quotes except when necessary vs always put quotes anyway) and ended up with both.

File: core/core/policies/unsupported_intent_policy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fparga) [reply core/core/policies/unsupported_intent_policy.py:39]: I wasn't saying it was your fault :stuck_out_tongue:

File: core/domain/domain.en.yml (hunk 1 above)
Comment (fparga) [reply core/domain/domain.en.yml:1248]: Ok :shrug:

**Pull request merged**